src/controllers/challenge.controller.js

```diff
@@ -1,17 +1,16 @@
 // Mock repository
 const data = require("../../ex.json");
 
-const getRandomChallenge = (req, res) => {
-  const question = data[Math.floor(Math.random() * data.length)];
-  res.send(question);
-};
-
-const getChallengeById = (req, res) => {
-  const question = data.find((obj) => obj.id === req.params.id);
-  res.send(question);
+const getChallenge = (req, res) => {
+  let options = [...data];
+  if (req.query.level) {
+    options = options.filter(
+      (question) => question.level === parseInt(req.query.level)
+    );
+  }
```

Comment:
This looks really off to me. I think it is because of how the callback is written. Normally you'd see something like:

```js
  options = options.filter(question => question.level === parseInt(req.query.level)
```

or even

```js
  options = options.filter(question => (
    question.level === parseInt(req.query.level)
  );
```

Reply:
You can add ()'s around the `question` prop if you want. If it makes you feel better about yourself.

Reply:
Prettier did it